code/filesystem/locator/abstract.php

```diff
@@ -0,0 +1,241 @@
+     *
+     * Called from {@link __construct()} as a first step of object instantiation.
+     *
+     * @param  ObjectConfig $config  An optional ObjectConfig object with configuration options.
+     * @return void
+     */
+    protected function _initialize(ObjectConfig $config)
+    {
+        $config->append(array(
+            'path_templates' => array()
+        ));
+
+        parent::_initialize($config);
+    }
+
+    /**
+     * Get the locator name
+     *
+     * @return string The stream name
+     */
+    public static function getName()
+    {
+        return static::$_name;
+    }
+
+    /**
+     * Locate the resource based on a url
+     *
+     * @param  string $url  The resource url
+     * @return string|false  The physical file path for the resource or FALSE if the url cannot be located
+     */
+    final public function locate($url)
+    {
+        $result = false;
+
+        if(!isset($this->__location_cache[$url]))
+        {
+            $info = $this->parseUrl($url);
+
+            //Find the file
+            foreach($this->getPathTemplates($url) as $template)
+            {
+                $path = str_replace(
+                    array('<Package>'     , '<Path>'     ,'<File>'      , '<Format>'     , '<Type>'),
+                    array($info['package'], $info['path'], $info['file'], $info['format'], $info['type']),
+                    $template
+                );
+
+                if ($results = glob($path))
+                {
+                    foreach($results as $file)
+                    {
+                        if($result = $this->realPath($file)) {
+                            break (2);
+                        }
+                    }
+                }
+            }
+
+            $this->__location_cache[$url] = $result;
+        }
+
+        return $this->__location_cache[$url];
+    }
+
+    /**
+     * Parse the url
+     *
+     * @param  string $url The language url
+     * @return array
+     */
+    public function parseUrl($url)
+    {
+        $scheme   = parse_url($url, PHP_URL_SCHEME);
+        $domain   = parse_url($url, PHP_URL_HOST);
+        $basename = pathinfo($url, PATHINFO_BASENAME);
+
+        $parts = explode('.', $basename);
+
+        if(count($parts) == 3)
+        {
+            $type  =  array_pop($parts);
+            $format = array_pop($parts);
+            $file   = array_pop($parts);
+        }
+        else
+        {
+            $type   = '*';
+            $format = array_pop($parts);
+            $file   = array_pop($parts);
+        }
+
+        $path = str_replace(array($scheme.'://', $scheme.':'), '', dirname($url));
+        $parts = explode('/', $path);
+
+        $info = array(
+            'type'    => $type,
+            'domain'  => $domain,
+            'package' => array_shift($parts),
+            'path'    => implode('/', $parts),
+            'file'    => $file,
+            'format'  => $format ?: '*',
```

Comment:
Not always. The implementation allows you to not specify a format in which case a wildcard will be added and the locator will use the first format it finds.